File: LAB 7/4.m
clc; clear;

% Point in world coordinates
P = [5; 5; 5; 1];

% Simple camera translation (moving world opposite)
Tx = -2; Ty = -3; Tz = -4;

T = [1 0 0 Tx;
     0 1 0 Ty;
     0 0 1 Tz;
     0 0 0 1];

P_view = T * P;

disp('View Coordinate:');
disp(P_view);

File: LAB 7/4.py
import numpy as np

P = np.array([5, 5, 5, 1])

T = np.array([
    [1, 0, 0, -2],
    [0, 1, 0, -3],
    [0, 0, 1, -4],
    [0, 0, 0, 1]
])

P_view = T @ P
print("View Coordinate:", P_view)